core/src/main/java/edu/ucr/cs/riple/core/AnalysisMode.java

```diff
@@ -0,0 +1,158 @@
+              report.tag(REJECT);
+            }
+          });
+    }
+  },
+
+  /**
+   * Guarantees that no error will happen on downstream dependencies in the result of changes in
+   * upstream.
+   */
+  STRICT {
+    @Override
+    public void tag(Config config, GlobalAnalyzer analyzer, Collection<Report> reports) {
+      reports.forEach(
+          report -> {
+            // Check for destructive methods.
+            if (report.containsDestructiveMethod(analyzer)) {
+              report.tag(REJECT);
+              return;
+            }
+            // Apply if effect is less than 1.
+            if (report.localEffect < 1) {
+              report.tag(APPROVE);
+              return;
+            }
+            // Discard.
+            report.tag(REJECT);
+          });
+    }
+  },
+
+  /**
+   * Experimental: Upper bound of number of errors on downstream dependencies will be considered.
+   */
+  UPPER_BOUND {
+    @Override
+    public void tag(Config config, GlobalAnalyzer analyzer, Collection<Report> reports) {
+      reports.forEach(
+          report -> {
+            if (report.localEffect + report.getUpperBoundEffectOnDownstreamDependencies() < 1) {
+              report.tag(APPROVE);
+            } else {
+              report.tag(REJECT);
+            }
+          });
+    }
+  },
+
+  /**
+   * Lower bound of number of errors on downstream dependencies will be considered. Default mode if
+   * downstream dependencies analysis is enabled.
+   */
+  LOWER_BOUND {
+    @Override
+    public void tag(Config config, GlobalAnalyzer analyzer, Collection<Report> reports) {
+      reports.forEach(
+          report -> {
+            if (report.localEffect + report.getLowerBoundEffectOnDownstreamDependencies() < 1) {
+              report.tag(APPROVE);
+            } else {
+              report.tag(REJECT);
+            }
+          });
+    }
+  };
+
+  /**
+   * Tags reports based on the analysis mode.
+   *
+   * @param config Annotator config.
+   * @param analyzer Downstream dependency instance.
+   * @param reports Reports to be processed.
+   */
+  public abstract void tag(Config config, GlobalAnalyzer analyzer, Collection<Report> reports);
+
+  /**
+   * Parses the received option and returns the corresponding {@link AnalysisMode}. Can only be one
+   * of [default|upper_bound|lower_bound|strict] values.
+   *
+   * @param downStreamDependenciesAnalysisActivated if true, downstream dependency analysis is
+   *     activated.
+   * @param mode passed mode.
+   * @param useDefault if true, no value is provided by the user and the default mode will be
+   *     returned. (If downstream dependency analysis is activated default is {@link
+   *     AnalysisMode#LOWER_BOUND}, otherwise the default is {@link AnalysisMode#LOCAL}).
+   * @return the corresponding {@link AnalysisMode}.
+   */
+  public static AnalysisMode parseMode(
+      boolean downStreamDependenciesAnalysisActivated, @Nullable String mode, boolean useDefault) {
```

Comment:
Why do we need `useDefault` here or a `@Nullable` `mode`, if simply passing `mode = "default"` will have the same effect. Looks like the logic and interface of this method can be greatly simplified.